### ISSUE TITLE
[Ops] Transpose: Support for transpose over -3 dim when it's not tile-dim aligned

### ISSUE THREAD

**Issue by nvukobratTT** [tenstorrent/tt-mlir#558]

**Comment by sdjordjevicTT** [comment]
@nvukobratTT @mtopalovicTT do we have some action needed here?

**Comment by nvukobratTT** [comment]
> @nvukobratTT @mtopalovicTT do we need some action here?

Yes, we'll need it. 

Currently, we're using a TensorFlow variation of ResNet 50 to by pass the issue. However, without transposing over -3 I'm not certain if can we push any conv-based model from PyTorch. More precisely, we need to transpose over -3 to convert channel first into channel last conv2d.

**Comment by sdjordjevicTT** [comment]
Do we have a repro MLIR for it? Do we need to create some issues on the metal side?

**Comment by nvukobratTT** [comment]
> Do we have a repro MLIR for it? Do we need to create some issues on the metal side?

@LPanosTT will create an issue today for this problem on tt-metal repo with TT-NN as repro. I'm not sure do you see any value of MLIR repro as well, as we have TT-NN already? :))

**Comment by sdjordjevicTT** [comment]
Nope, I am just trying to understand do you need anything urgent from us regarding this ongoing issue? :)

**Comment by nvukobratTT** [comment]
> Nope, I am just trying to understand do you need anything urgent from us regarding this ongoing issue? :)

No worries, I already discussed it with Mladen, this one is not P0 at the moment, but has the potential to become one in upcoming weeks :))

**Comment by sdjordjevicTT** [comment]
Great, thanks for clarification! :)

**Comment by dgolubovicTT** [comment]
Hi, I have experienced the need for transpose along -3 dimension while bringing up LLama3B. Namely, initial graph contains transposes along -3 dim:
<img width="1234" alt="image" src="https://github.com/user-attachments/assets/fa06740e-929d-46f5-987d-354745399a50">


Currently, those transposes are decomposed after optimization passes to following ops: 2xpad_tile, matmul, vslice, vstack and 2x narrow. I ran the transpose test changing one of the transpose axes to -3 and I get that:

<img width="652" alt="image" src="https://github.com/user-attachments/assets/3f12da6d-fe64-468f-bd74-b3c9d813a9be">

Since this pad_tile is not supported end to end, and vslice and vstack are depricated, I would say that transpose along -3 is a blocker for further bringup of LLama 3B.

**Comment by nvukobratTT** [comment]
Thanks for the details @dgolubovicTT! This now became a blocker for Llama 3B, and near future blocker for ResNet 50. 

@dgolubovicTT when you open up issues on tt-metal, please link it here as a ref 💪

**Comment by sdjordjevicTT** [comment]
Do we have an issue created on the metal side?

**Comment by nvukobratTT** [comment]
> Do we have an issue created on the metal side?

https://github.com/tenstorrent/tt-metal/issues/13005

**Comment by nvukobratTT** [comment]
I'm updating blocker field as well

**Comment by sdjordjevicTT** [comment]
We just consumed a new metal in TT-MLIR, @nvukobratTT @dgolubovicTT can we verify if this works from the Forge side? :)

**Comment by dgolubovicTT** [comment]
I will test it as soon as we uplift new tt-mlir...

**Comment by dgolubovicTT** [comment]
Transpose on -3 works after uplifting new metal in TT-MLIR.
